src/unitxt/operators.py

```diff
@@ -213,6 +213,97 @@ def process(
         return flatten_dict(instance, parent_key=self.parent_key, sep=self.sep)
 
 
+class WholeInputFormatter(StreamInstanceOperator):
+    r"""Generates the whole input to the model, from constant strings that are given as args, and from values found in specified fields of the instance.
+
+    WholeInputFormatter expects its input instance to have been processed by a Template StreamInstanceOperator
+    (from templates.py), or any extension thereof, and hence to contain a field named "source", and to have
+    been processed by RenderInstruction StreamInstanceOperator (from renderer.py) and hence to contain a non None
+    field "instruction".
+    Demos are assumed to be included as well, in a field whose name is specified by argument 'demos_field'. That
+    field is assumed to contain a list of dicts, each containing keys: "source" and "target" of a single demo.
+    And finally, system_prompt is also assumed to be included
+    in the instance, in a field whose name is specified by argument 'system_prompt'.
+    The value in field "source" of the input instance describes (verbalizes) original values in the instance (as read
+    from the source dataset), in the context of the underlying task.
+    WholeInputFormatter makes this (verbalization of the) input into a whole-input-to-the-model, by combining it
+    with instruction, demos, and system_prompt, read from the input instance, through both formatting args
+    formatting arguments.
+
+    Args:
+        to_field (str): the name of the field into which the formatted model's input is to be stored
+        demos_field (str): the name of the field that contains the demos, being dicts with "source" and "target" keys
+        system_prompt (str): the name of the field containing an introductory text and or xml tags
+        demo_format (str): formatting string for a single demo, combining fields source and target
+        general_format (str) overall output format, combinig system_prompt, instruction, demos, and source
+    """
+
```

Comment:
Important to add examples...

Reply:
they are very long.. please see in `test_operators` . Shall I copy over such long examples?

Reply:
I mean an example like this:

"demo_format": "Human: {source}\nAssistant:  {target}\n\n",
    "model_input_format": "{system_prompt}{instruction}\n{demos}\nHuman: {source}\nAssistant: "
    
    Will yield output like:
    
    
    ...

Reply:
Added example. Indeed not terribly long.

Reply:
Thanks. This format is a bit odd because the instruction appears after the demos. 

 model_input_format="{demos}User: {instruction}{source}\nAgent: ",
 
In general, we need to clarify the confusion between the instruction and system prompt.  
Today, it's a mess.  The input parameter to the standard recipe is called instruction, but it is used to pass the system prompt.  In the unit test, instructions are task instructions like "solve the math exercisers".

Let's discuss it with Elron, but for now , let's give this example:
demo_format="Input: {source}\nOutput: {target}\n\n",
 model_input_format="Instruction: {instruction}\n{demos}Input: {source}\nOutput: "

Reply:
OK. changed the example.